src/win.rs

```diff
@@ -344,3 +350,26 @@ pub fn windows_interfaces(display: InterfaceDisplay) -> Result<Vec<String>, Box<
             .collect()
     })
 }
+
+/// List all the network interfaces available on the system by their indexes
+pub fn windows_interfaces_by_index(
+    display: InterfaceDisplay,
+) -> Result<types::IfacesByIndex, Box<dyn std::error::Error>> {
+    let interfaces = win_explore_adapters();
```

Comment:
Because this function returns `Box<dyn std::error::Error>>`, you can add a `?` after the call to `win_explore_adapters` (ie, `win_explore_adapters()?;`) and this will bubble up the error. This avoids the `match` below without any behavior change

Reply:
ahh, wow that's really cool!  Changed!